src/Collection/ImmutableObjectCollection.php

```diff
@@ -12,15 +12,17 @@
 
 use Gamee\Collections\Iterator\ObjectIterator;
 
-class ImmutableObjectCollection extends ObjectIterator
+abstract class ImmutableObjectCollection extends ObjectIterator
 {
 
 	/**
 	 * @throws \InvalidArgumentException
 	 */
 	public function addItem($item): self
 	{
-		if (!$item instanceof $this->getItemType()) {
+		$className = $this->getItemType();
```

Comment:
Could that be a `$itemClassName` instead of `$className`? I think it would make much more sense.